logins-api/android/library/src/main/java/org/mozilla/sync15/logins/ServerPassword.kt

```diff
@@ -47,5 +50,53 @@ class ServerPassword (
 
     val usernameField: String? = null,
     val passwordField: String? = null
-)
+) {
 
+
+    companion object {
+        fun fromJSON(jsonObject: JSONObject): ServerPassword {
+
+            fun stringOrNull(s: String): String? {
+                try {
+                    return jsonObject.getString(s)
+                } catch(e: JSONException) {
+                    return null
+                }
+            }
+
+            return ServerPassword(
+                    id = jsonObject.getString("id"),
+
+                    hostname = jsonObject.getString("hostname"),
+                    password = jsonObject.getString("password"),
+                    username = stringOrNull("username"),
```

Comment:
This is what `optString` is for: https://developer.android.com/reference/org/json/JSONObject.html#optString(java.lang.String,%20java.lang.String)

Reply:
For some reason I forgot that Java strings could be null (Kotlin ones can't, FWIW)